lib/Core/BuildEngine.cpp

```diff
@@ -199,7 +202,13 @@ class BuildEngineImpl : public BuildDBDelegate {
 
     bool isComplete(const BuildEngineImpl* engine) const {
       return state == StateKind::Complete &&
-        result.builtAt == engine->getCurrentTimestamp();
+        result.builtAt == engine->getCurrentIteration();
+    }
+    
+    void setComputing(const BuildEngineImpl* engine) {
+      assert(isInProgressWaiting());
+      state = StateKind::InProgressComputing;
+      result.relativeStart = Clock::durationSince(engine->buildStartedTimestamp);
```

Comment:
Why record relative start, instead of just getting a timestamp here, and then subtracting? Basically we currently do:
```
relativeStart = now() - buildStart
...
duration = now() - (relativeStart + buildStart)
```
instead of
```
start = now()
duration = now() - start
```

Reply:
Ah, I see, you want to store relative start in the database. I wasn't expecting that, but can see why it might be useful (definitely nice for visualization)

I think I would still prefer if we put the absolute timestamp in the ruleInfo, and then did the math when we set the result properties, I think.

Reply:
I agree with keeping the absolute timestamp in the ruleinfo.   I do wonder if it would make sense to store the timestamps in the database too.  With them we can do analysis on the scheduling of the tasks, and it is easy to derive the duration from the two values.

Reply:
I like that idea, it would also let us clearly see things like when certain items were rebuilt

Reply:
I think that depends what the timing information is being used for. I imagine that we'll use it to find out more about the events that happened during a build. With that in mind, the relative start is the only important information, it doesn't matter when in time the whole build was executed.
But if you think it makes sense to have the actual timestamp in the database, we can do that too. But then we also need to keep in mind what happens if the date/time gets changed/synchronized and what that means for the values.

Reply:
Since the duration is easily derived, if we are storing both start and stop I think there is no good reason to avoid storing the absolute values, since it is strictly more info. Even if it only helps with debugging, that still has non-negative value

Reply:
Yep, that's true. I'll change it